tools/landmark_registration_ls/landmark_registration_ls.xml

```diff
@@ -0,0 +1,33 @@
+<tool id="ip_landmark_registration_ls" name="Landmark Registration" version="0.0.1">
+    <description>using least squares</description>
+    <requirements>
+        <requirement type="package" version="1.2.1">scipy</requirement>
+        <requirement type="package" version="0.23.4">pandas</requirement>
+        <requirement type="package" version="1.15.2">numpy</requirement>
+    </requirements>
+    <command><![CDATA[ 
+         python '$__tool_directory__/landmark_registration_ls.py'
+         '$fn_pts1'
+         '$fn_pts2'
+         '$fn_tmat'
+]]></command>
+    <inputs>
+          <param name="fn_pts1" type="data" format="tabular" label="Path to tab-separated file with src points" />
+          <param name="fn_pts2" type="data" format="tabular" label="Path to tab-separated file with dst points" />
+    </inputs>
+    <outputs>
+       <data format="tabular" name="fn_tmat" />
+    </outputs>
+    <tests>
+        <test>
+            <param name="fn_pts1" value="points1.tsv"/>
+            <param name="fn_pts2" value="points2.tsv"/>
+            <output name="fn_tmat" value="tmat.tsv" ftype="tabular" compare="diff" lines_diff="6"/>
```

Comment:
If you have diff lines here with 6 but your file has only 3 lines you are basically testing nothing.
What was the problem here? Rounding differences? Maybe you can use assert based tests?

Reply:
Yes, there are always numerical errors (finite precision of computations).

I copied this line from the existing xml of the tool landmark_registration. I'd mentioned in the email that the automatic testing is very limited. You said it mainly/only tests the wrapper, parameters, outputs...

Functional tests have been done off-line.

Reply:
> I copied this line from the existing xml of the tool landmark_registration. I'd mentioned in the email that automatic testing is very limited. You said it mainly/only tests the wrapper, parameters, outputs...

Yes, but this test is really doing nothing. So if for some reason the file is empty is test would also pass. Have a look at https://docs.galaxyproject.org/en/latest/dev/schema.html#id79 we can even use regex is such tests to adjust for numerical instability.